From 9079079ec0dac15700400a8c86b12595e9524901 Mon Sep 17 00:00:00 2001
From: Aurimas Liutikas <aurimas@google.com>
Date: Fri, 10 Feb 2017 16:23:49 -0800
Subject: [PATCH] Fix compiler warnings about unused variables in wire_format.h

---
 src/google/protobuf/wire_format.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/google/protobuf/wire_format.h b/src/google/protobuf/wire_format.h
index 67afc77..5e9aca5 100644
--- src/google/protobuf/wire_format.h
+++ src/google/protobuf/wire_format.h
@@ -326,5 +326,8 @@ inline void WireFormat::VerifyUTF8StringNamedField(
 #ifdef GOOGLE_PROTOBUF_UTF8_VALIDATION_ENABLED
   WireFormat::VerifyUTF8StringFallback(data, size, op, field_name);
+#else
+  // Avoid the compiler warning about unused variables.
+  (void)data; (void)size; (void)op; (void)field_name;
 #endif
 }
 
